emission/analysis/modelling/trip_model/greedy_similarity_binning.py

```diff
@@ -207,14 +207,22 @@ def _find_matching_bin_id(self, trip_features: List[float]) -> Optional[str]:
         finds an existing bin where all bin features are "similar" to the incoming
         trip features.
 
-        :param trip_features: feature row for the incoming trip
+        :param trip_features: feature row for the incoming trip. 
+                            takes the form [orig_lat, orig_lon, dest_lat, dest_lon]
         :return: the id of a bin if a match was found, otherwise None
         """
         for bin_id, bin_record in self.bins.items():
-                matches_bin = all([self.metric.similar(trip_features, bin_sample, self.sim_thresh,self.clusteringWay)
-                    for bin_sample in bin_record['feature_rows']])
-                if matches_bin:
-                    return bin_id
+            if self.clusteringWay == 'origin':
```

Comment:
Moving `clusteringWay` parameter based decision making from `od_similarity`  here. This helps generalize similarity calculations. Earlier, complete features list was passed to `similar` in every scenario and  decisions taken there.

Reply:
As I said before:
> Note that it might be easier to add explanations as code comments, like we used to in the old days.

https://github.com/e-mission/e-mission-server/pull/933#discussion_r1300940088

> While this is better than nothing - it requires people to use git blame to find the commit, then find the related PR and scroll down to find these comments. And since we will be resolving them, they would all be hidden, so they would not be searchable anyway.

https://github.com/e-mission/e-mission-eval-private-data/pull/37#discussion_r1302993000

Please put long-term code comments into the code